apps/site/lib/site_web/controllers/event_controller.ex

```diff
@@ -121,4 +125,44 @@ defmodule SiteWeb.EventController do
   defp decode_ampersand_html_entity(string) do
     String.replace(string, "&amp;", "&")
   end
+
+  @doc "Returns a list of years with which we can filter events.
+  Defaults to the current datetime if no assigns
+  "
+  @spec year_options(Plug.Conn.t()) :: %Range{:first => Calendar.year(), :last => Calendar.year()}
+  def year_options(%{assigns: %{date: %{year: year}}}) when is_integer(year) do
+    do_year_options(year)
+  end
+
+  def year_options(_) do
+    %{year: year} = Util.now()
+    do_year_options(year)
+  end
+
+  @spec do_year_options(Calendar.year()) :: %Range{
+          :first => Calendar.year(),
+          :last => Calendar.year()
+        }
+  defp do_year_options(year), do: Range.new(year - 4, year + 1)
+
+  @spec get_years_for_selection(Plug.Conn.t()) :: list
+  defp get_years_for_selection(%{assigns: %{events: events}} = conn) do
+    year_range = year_options(conn)
+
+    Enum.map(year_range, fn year ->
+      events_for_that_year =
+        if year == conn.assigns.year do
+          events
+        else
+          Repo.events_for_year(year)
```

Comment:
Neat approach. Since you end up fetching the events for each year anyway I wonder if it might make more sense to modify the `assign_events` plug to include this function's work. As in,
- do the `Repo.events_for_year` for each of the 5 years
- assign `conn.assigns.events` based on the events corresponding to `conn.assigns.year`, (_or_ maybe assign a variable that's a map of year & events? hmm)
- assign the years for selection in the manner you do here.